plugin/trino-hive/src/main/java/io/trino/plugin/hive/AcidInfo.java

```diff
@@ -265,23 +277,29 @@ public Builder addOriginalFile(Path originalFilePath, long originalFileLength, i
             return this;
         }
 
+        public Builder setOrcAcidVersionValidated(boolean orcAcidVersionValidated)
+        {
+            this.orcAcidVersionValidated = orcAcidVersionValidated;
+            return this;
+        }
+
         public AcidInfo buildWithRequiredOriginalFiles(int bucketId)
         {
             checkState(
                     bucketId > -1 && bucketIdToOriginalFileInfoMap.containsKey(bucketId),
                     "Bucket Id to OriginalFileInfo map should have entry for requested bucket id: %s",
                     bucketId);
             List<DeleteDeltaInfo> deleteDeltas = ImmutableList.copyOf(deleteDeltaInfos);
-            return new AcidInfo(partitionLocation.toString(), deleteDeltas, bucketIdToOriginalFileInfoMap.get(bucketId), bucketId);
+            return new AcidInfo(partitionLocation.toString(), deleteDeltas, bucketIdToOriginalFileInfoMap.get(bucketId), bucketId, orcAcidVersionValidated);
         }
 
         public Optional<AcidInfo> build()
         {
             List<DeleteDeltaInfo> deleteDeltas = ImmutableList.copyOf(deleteDeltaInfos);
-            if (deleteDeltas.isEmpty()) {
+            if (deleteDeltas.isEmpty() && orcAcidVersionValidated) {
```

Comment:
maybe we dont need the `&&` here if the diff is in delete delta files?

in any case please add a comment

Reply:
Maybe - but this is safer. And I am not sure if delete deltas are only issue.

Reply:
I added a comment.